### ISSUE TITLE
"keys.map is not a function" crash when supplying a string to asKeyStore() method.

### ISSUE THREAD

**Issue by sc0ttdav3y** [cisco/node-jose#279]
Hi,

I'm writing a node CLI tool with the latest node-jose 1.1.3 and node 13.8.0. 

When I try to create a key store using the `asKeyStore` method, I get the following error:

```
TypeError: keys.map is not a function
    at Object.JWKStore.asKeyStore (./node_modules/node-jose/lib/jwk/keystore.js:611:15)
```

The calling code is like this:

```
let jwks = fs.readFileSync(filename); // a JSON string
let keyStore = await jose.JWK.asKeyStore(jwks); // crash!
```

If I modify my code to pass an array of keys instead, it works fine:

```
let jwks = JSON.parse(fs.readFileSync(filename)); // an object
let keyStore = await jose.JWK.asKeyStore(jwks.keys); // works fine
```

Perhaps I'm using it wrong, but it looks like it's supposed to support a JWKS string as an input.

Anyway, it looks like `keystore.js:610` could use a `if (Array.isArray(keys))` assertion.   

Cheers, Scott

**Comment by phish108** [comment]
in [node-jose-tools](/phish108/node-jose-tools) I created a small wrapper to ensure the correct behavior. Please feel free to have a look at its addkey tool. 

There are more side cases with JWK, so I won’t consider this a bug.

**Comment by sc0ttdav3y** [comment]
Thanks for your reply.

> ...so I won’t consider this a bug because you provide the wrong input to the function.

In my original message I stated that I was providing a `String`, and the docs say the function accepts a `String`, so this comment didn't initially make any sense to me.

A bit more digging reveals that Node's `fs.readFileSync()` function returns either a `String` or a `Buffer`, and in this case it returned a `Buffer`. It seems you were more observant than me and knew this all along.

In any event, I worked around it and hopefully this ticket helps others in future with the same problem.
